### PR TITLE
cobertura

### PR DESCRIPTION
**Pull request by finomen**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -94,6 +95,15 @@ func main() {
 		}
 	}
 
+	// Xml report, if requested.
+	if *cobertura != "" {
+		err := createCoberturaReport(*cobertura, fileData, report)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: could not create xml report: %s\n", err)
```
Comment (Javier-varez) [inline main.go:102]: You could also use `log.Fatalf` here

File: lcov.go [hunk 2]
```diff
@@ -135,9 +135,13 @@ func parseBRDARecord(value string) (lineNo int, status BranchStatus, err error)
...
-	hitCount, err := strconv.ParseInt(values[3], 10, 64)
-	if err != nil {
-		return 0, 0, fmt.Errorf("can't parse branch record: %s", err)
+	var hitCount int64 = 0
+
+	if values[3] != "-" {
+		hitCount, err = strconv.ParseInt(values[3], 10, 64)
+		if err != nil {
+			return 0, 0, fmt.Errorf("can't parse branch record: %s", err)
+		}
```
Comment (Javier-varez) [inline lcov.go:145]: I have no idea what this does, a comment here would help me (or someone in the future to get the necessary context for this lcov parsing).

File: cobertura.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		parent := path.Dir(file.Name)
+		pkgName := strings.ReplaceAll(parent, "/", ".")
+
+		pkg, ok := packages[pkgName]
+
+		if !ok {
+			pkg = &XmlPackage{
+				Name:       pkgName,
+				LineRate: float64(data.FileData(file.Name).LineCoverage().P()) / 100.0,
+				BranchRate: float64(data.FileData(file.Name).BranchCoverage().P()) / 100.0,
```
Comment (Javier-varez) [inline cobertura.go]: Doesn't this refer to the file rather than to the package?

File: main.go (hunk 1 above)
Comment (finomen) [reply main.go:102]: All other checks using printf/exit

Comment (Javier-varez) [reply main.go:102]: Ok, not a strong opinion, just a suggestion.

**Review by Javier-varez — APPROVED** [review]

**Pull request merged**